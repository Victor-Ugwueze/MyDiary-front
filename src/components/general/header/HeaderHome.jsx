// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const HeaderHome = props => (
      <header className="home-header">
          <img className="loading-indicator" src="/images/icons/loading_spinner3.gif" />
            <div className="container-fluid">
                <div className="brand">
                    <div>
                            My Dairy
                    </div>
                    <span className="icon-bar hide" id="menu-icon"></span>
                </div>
                <nav>
                    <ul className="navbar">
                        <li className="nav-link">
                            <a href="index.html">
                                Home
                            </a>
                        </li>
                        <li className="nav-link">
                            <a href="#"
                                onClick={props.openModal}
                                data-target="sign-up-login"
                                data-toggle="login"
                            >
                                login
                            </a>
                        </li>
                        <li className="nav-link" >
                            <a href="#"
                                data-toggle="signup"
                                onClick={props.openModal}
                                data-target="sign-up-login">
                                Sign up for Free
                            </a>
                        </li>
                    </ul>
                </nav>
            </div>
            <div id="intro">
                <div id="home-text">
                    <h1>Welcome to My Diary</h1>
                    <p>A place to pen down all your toughts</p>
                    <a href="#how-it-works" className="btn btn-primary how-it-works-btn">How It Works</a>
                </div>

            </div>
        </header>
);

HeaderHome.propTypes = {
  openModal: PropTypes.func,
};

export default HeaderHome;
