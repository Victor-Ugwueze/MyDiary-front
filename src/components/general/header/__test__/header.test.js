// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// components
import HeaderHome from '../HeaderHome';


configure({
  adapter: new Adapter(),
});

describe('Renders the header of landing page', () => {
  it('should render herder section of landing page', () => {
    const component = shallow(
      <HeaderHome />
    );
    expect(component).toMatchSnapshot();
  });
});
