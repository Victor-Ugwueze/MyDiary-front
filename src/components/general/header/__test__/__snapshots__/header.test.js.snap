// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the header of landing page should render herder section of landing page 1`] = `
<header
  className="home-header"
>
  <img
    className="loading-indicator"
    src="/images/icons/loading_spinner3.gif"
  />
  <div
    className="container-fluid"
  >
    <div
      className="brand"
    >
      <div>
        My Dairy
      </div>
      <span
        className="icon-bar hide"
        id="menu-icon"
      />
    </div>
    <nav>
      <ul
        className="navbar"
      >
        <li
          className="nav-link"
        >
          <a
            href="index.html"
          >
            Home
          </a>
        </li>
        <li
          className="nav-link"
        >
          <a
            data-target="sign-up-login"
            data-toggle="login"
          >
            login
          </a>
        </li>
        <li
          className="nav-link"
        >
          <a
            data-target="sign-up-login"
            data-toggle="signup"
            href="#"
          >
            Sign up for Free
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    id="intro"
  >
    <div
      id="home-text"
    >
      <h1>
        Welcome to My Diary
      </h1>
      <p>
        A place to pen down all your toughts
      </p>
      <a
        className="btn btn-primary how-it-works-btn"
        href="#how-it-works"
      >
        How It Works
      </a>
    </div>
  </div>
</header>
`;
